src/standard-live-connect.ts

```diff
@@ -83,35 +85,53 @@ function _getInitializedLiveConnect (liveConnectConfig: LiveConnectConfig): ILiv
 
 function _standardInitialization (liveConnectConfig: LiveConnectConfig, externalStorageHandler: StorageHandler, externalCallHandler: CallHandler, eventBus: EventBus): ILiveConnect {
   try {
-    const callHandler = new WrappedCallHandler(externalCallHandler, eventBus)
+    // TODO: proper config validation
     const validLiveConnectConfig = removeInvalidPairs(liveConnectConfig, eventBus)
-    const configWithPrivacy = mergeObjects(validLiveConnectConfig, privacyConfig(validLiveConnectConfig))
-    errorHandler.register(configWithPrivacy, callHandler, eventBus)
-    const storageStrategy = configWithPrivacy.privacyMode ? StorageStrategies.disabled : configWithPrivacy.storageStrategy
-    const storageHandler = WrappedStorageHandler.make(storageStrategy, externalStorageHandler, eventBus)
-    const reducer = (accumulator, func) => accumulator.combineWith(func(accumulator.data, storageHandler, eventBus))
 
-    const enrichers = [pageEnrich, identifiersEnrich]
-    const managers = [idResolve, decisionsResolve]
+    const stateBuilder = new EnrichmentContext({
+      ...validLiveConnectConfig,
+      identifiersToResolve: validLiveConnectConfig.identifiersToResolve || [],
+      contextSelectors: validLiveConnectConfig.contextSelectors || "",
+      contextElementsLength: validLiveConnectConfig.contextElementsLength || 0,
+      callHandler: externalCallHandler,
+      storageHandler: externalStorageHandler,
+      eventBus
+    })
 
-    const enrichedState = enrichers.reduce(reducer, new StateWrapper(configWithPrivacy, eventBus))
-    const postManagedState = managers.reduce(reducer, enrichedState)
+    const enrichedState = stateBuilder
```

Comment:
This is what all of the enricher refactor is for.
If you forget anything or order steps incorrectly, the compiler will beat you